src/App.tsx

```diff
@@ -25,16 +35,54 @@ function App() {
         <p>
           <input type='text' id="auth-code" placeholder='Paste authentication code here' />
           <button onClick={() => {
-            let code = (document.getElementById("auth-code") as HTMLInputElement).value;
-            invoke('finalize_auth', { code });
+            let code: string = (document.getElementById("auth-code") as HTMLInputElement).value;
+            invoke('finalize_auth', { code })
+              .catch((err) => console.error(err))
+              .then(() => props.setState(1));
           }}>Submit</button>
         </p>
         <p>
           Edit <code>App.tsx</code> and save to test HMR updates.
         </p>
       </header>
+    </div >
+  )
+};
+
+const MainPage = (props: BaseParams) => {
+
+  useEffect(() => {
+    invoke('upsert_template').catch((err) => { console.error(err) });
+  }, []);
+
+  return (
+    < div className="App" >
+      <div className='nav'>
+      </div>
     </div>
   )
+};
+
+const ErrorPage = () => {
+  return (
+    <div>Error, this page occurs when the app attemps to swap to a page that doesn't exist. Please open an issue if you ever see this page</div>
+  )
+}
+
+function App() {
+
+  const [state, setState] = useState(0);
+
+  //depending on the state of the app (set through an enum, the app will display a specific page)
+  switch (state) {
+    case 0:
```

Comment:
for tracking the current page, what do you think about using a constant instead?

For example, outside of any react function you could define something like:

```js
const PAGE_AUTH = "PAGE_AUTH";
const PAGE_MAIN = "PAGE_MAIN";
```
While this approach may seem overly verbose, it could make the application easier to understand as you are always setting the state to something meaningful, you don't have to resolve the number back to what the routing function considers it to be.